src/vm/ssa.nit

```diff
@@ -0,0 +1,1140 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Copyright 2015 Julien Pagès <julien.pages@lirmm.fr>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Compute Single-Static Assignment form from an AST
+module ssa
+
+import variables_numbering
+import astbuilder
+
+redef class VirtualMachine
```

Comment:
You should try to detach ssa from the vm: you introduce unwanted coupling (thus complexity) and SSA is a nice thing to hae independently of the vm (compiler, static analysis tools, etc).